integration-tests/test_connection.py

```diff
@@ -6,12 +6,43 @@
 :upstream: Yes
 """
 
+import configparser
+import os
+import typing
+
 import pytest
 
+if typing.TYPE_CHECKING:
+    import pytest_client_tools.insights_client
+
 # the tests need a valid connection to insights so therefore the subman registration
 pytestmark = pytest.mark.usefixtures("register_subman")
 
 
+def _is_using_proxy(
+    insights_config: "pytest_client_tools.insights_client.InsightsClientConfig",
+) -> bool:
+    for key in os.environ.keys():
+        if key.lower() == "https_proxy":
+            return True
+
+    try:
+        config_proxy = insights_config.proxy
+        if config_proxy != "":
+            return True
+    except KeyError:
+        return False
```

Comment:
shouldn't this be a simple `pass` (or `contextlib.suppress()`)? otherwise a missing `proxy` entry in `insights-client.conf` (which is the default) will cause `KeyError` to be raised, and thus `False` to be returned unconditionally, without trying to read `rhsm.conf`